dagger/dag_creator/airflow/operators/spark_submit_operator.py

```diff
@@ -85,40 +86,101 @@ def get_cluster_id_by_name(self, emr_cluster_name, cluster_states):
+        Get the application ID of the Spark job
         """
-        cluster_id = self.get_cluster_id_by_name(self.cluster_name, ["WAITING", "RUNNING"])
-        emr_master_instance_id = self.emr_client.list_instances(ClusterId=cluster_id, InstanceGroupTypes=["MASTER"],
-                                                                InstanceStates=["RUNNING"])["Instances"][0][
-            "Ec2InstanceId"]
-
-        command_parameters = {"commands": [self.spark_submit_cmd]}
-        if self._execution_timeout:
-            command_parameters["executionTimeout"] = [self.get_execution_timeout()]
+        command = f"yarn application -list -appStates RUNNING | grep {application_name}"
 
         response = self.ssm_client.send_command(
             InstanceIds=[emr_master_instance_id],
             DocumentName="AWS-RunShellScript",
-            Parameters= command_parameters
+            Parameters={"commands": [command]}
         )
+
         command_id = response['Command']['CommandId']
-        status = 'Pending'
-        status_details = None
-        while status in ['Pending', 'InProgress', 'Delayed']:
-            time.sleep(30)
-            response = self.ssm_client.get_command_invocation(CommandId=command_id, InstanceId=emr_master_instance_id)
-            status = response['Status']
-            status_details = response['StatusDetails']
-        self.log.info(
-            self.ssm_client.get_command_invocation(CommandId=command_id, InstanceId=emr_master_instance_id)[
-                'StandardErrorContent'])
-
-        if status != 'Success':
-            raise AirflowException(f"Spark command failed, check Spark job status in YARN resource manager. "
-                                   f"Response status details: {status_details}")
-
-    def on_kill(self):
-        self.log.info("Sending SIGTERM signal to bash process group")
-        os.killpg(os.getpgid(self.sp.pid), signal.SIGTERM)
+        time.sleep(10)  # Wait for the command to execute
+
+        output = self.ssm_client.get_command_invocation(
+            CommandId=command_id,
+            InstanceId=emr_master_instance_id
+        )
+
+        stdout = output['StandardOutputContent']
+        for line in stdout.split('\n'):
+            if application_name in line:
+                application_id = line.split()[0]
+                return application_id
+        return None
+
+    def kill_spark_job(self, emr_master_instance_id, application_id):
+        """
+        Kill the Spark job using YARN
+        """
+        kill_command = f"yarn application -kill {application_id}"
+        self.ssm_client.send_command(
+            InstanceIds=[emr_master_instance_id],
+            DocumentName="AWS-RunShellScript",
+            Parameters={"commands": [kill_command]}
+        )
+        raise AirflowException(
+            f"Spark job exceeded the execution timeout of {self._execution_timeout} seconds and was terminated.")
```

Comment:
Can this only happen if it's a timeout? What if we cancel the job on airflow UI?